FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,322 @@
+
+            if (nextTimestamp != null)
+            {
+                anchorList[(DateTimeOffset)nextTimestamp] = 0;
+            }
+
+            if (lastTimestamp != null)
+            {
+                // the value of anchor is resource counts less than the time stamp.
+                // Add 1 milliseond on the last resource time stamp.
+                anchorList[((DateTimeOffset)lastTimestamp).AddMilliseconds(1)] = totalCount;
+            }
+            else
+            {
+                anchorList[endTime] = totalCount;
+            }
+
+            return anchorList;
+        }
+
+        // get the lastUpdated time stamp of next resource.
+        private async Task<DateTimeOffset?> GetNextTimestamp(string resourceType, DateTimeOffset? start, DateTimeOffset end, bool isDescending, CancellationToken cancellationToken)
+        {
+            List<KeyValuePair<string, string>> parameters = new ()
+            {
+                new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"lt{end.ToInstantString()}"),
+                new KeyValuePair<string, string>(FhirApiConstants.PageCountKey, FhirApiPageCount.Single.ToString("d")),
+                new KeyValuePair<string, string>(FhirApiConstants.TypeKey, resourceType),
+                isDescending
+                ? new KeyValuePair<string, string>(FhirApiConstants.SortKey, LastUpdatedApiParameterDesc)
+                : new KeyValuePair<string, string>(FhirApiConstants.SortKey, LastUpdatedApiParameter),
+            };
+
+            if (start != null)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"ge{((DateTimeOffset)start).ToInstantString()}"));
+            }
+
+            var searchOptions = new BaseSearchOptions(null, parameters);
+
+            string fhirBundleResult = await _dataClient.SearchAsync(searchOptions, cancellationToken);
+
+            // Parse bundle result.
+            JObject fhirBundleObject;
+            try
+            {
+                fhirBundleObject = JObject.Parse(fhirBundleResult);
+            }
+            catch (JsonReaderException exception)
+            {
+                string reason = string.Format(
+                        "Failed to parse fhir search result for '{0}' with search parameters '{1}'.",
+                        searchOptions.ResourceType,
+                        string.Join(", ", searchOptions.QueryParameters.Select(parameter => $"{parameter.Key}: {parameter.Value}")));
+
+                _diagnosticLogger.LogError(reason);
+                _logger.LogInformation(exception, reason);
+                throw new FhirDataParseException(reason, exception);
+            }
+
+            List<JObject> fhirResources = FhirBundleParser.ExtractResourcesFromBundle(fhirBundleObject).ToList();
+
+            if (fhirResources.Any() && fhirResources.First().GetLastUpdated() != null)
```

Comment:
seems we don't need to check `fhirResources.First().GetLastUpdated() != null` here?

Reply:
yes, removed this check, thanks.